### ISSUE TITLE
Three.js source does not work as ES6 module

### ISSUE THREAD

**Issue by arodic** [mrdoob/three.js#14446]
Ideally, developers should be able to use Three.js in pure ES6 module form.

For example, the following should work:

```HTML
<script type="module">
import * as THREE from "./src/Three.js";
</script>
```

Instead, we get errors like this one:

```
Failed to load module script: The server responded with a non-JavaScript MIME type of "application/octet-stream". Strict MIME type checking is enforced for module scripts per HTML spec.
``` 

This happens because glsl shader chunks are not real modules and need to be processed during build step by rollup glsl plugin.

Fixing this would improve development experience and also help with #9403 and #9562

Suggested solutions:

1. Rollup ShadersChunks.js into ShaderChunksBuild.js separately before building the library.

2. Convert each individual glsl shader chunk into ES6 module (automatically).

3. Rewrite and maintain shader chunks as ES6 modules using template literals. For example:

```javascript
var encodings_fragment = glsl`
 gl_FragColor = linearToOutputTexel( gl_FragColor );
`;
export {encodings_fragment};
```

##### Three.js version

- [x ] Dev

##### Browser

- [x] All of them

##### OS

- [x] All of them

**Comment by arodic** [comment]
> 1. Rollup ShadersChunks.js into ShaderChunksBuild.js separately before building the library.

I already tested this approach and it works.

> 2. Convert each individual glsl shader chunk into ES6 module (automatically).

This should work but having shader chunks in both glsl and js form might confuse some developers.

> 3. Rewrite and maintain shader chunks as ES6 modules using template literals.

Problem with this one is that glsl template literals don't have syntax highlighting by default so developers would need to plugins for their code editors.

**Comment by arodic** [comment]
> Also, the glsl rollup plugin does some minor string clean up (removes comments).

@mrdoob yeah I'm not sure why it didn't remove comments in my PR but it appears there were other issues. It appeared that most people don't like partial builds in src/ directory. That leaves solution 3. and 4. only candidates for now.

Also, keep in mind that template literals can in fact strip out comments (they are actually functions). Moreover, comment stripping can be still be done during build process.

**Comment by looeee** [comment]
> Another solution suggested by @looeee: do nothing (or write better documentation): it will not be possible to import the three.js src directly in the browser, and people bundling the source themselves have to add plugins such as glslify to their build step (as they currently do)

To be clear, this was not so much a suggestion as an explanation of the current situation. I don't currently  have a strong opinion as to what the best approach is here.

**Comment by arodic** [comment]
I just pushed solution no.3 (template literals) to [shaderchunk-template](https://github.com/arodic/three.js/tree/shaderchunk-template) branch to illustrate what I meant. I don't expect this solution to be accepted.

This works as ES6 modules and does not require build step whatsoever but has several issues:

1. No syntax highlighting by default.
2. Not [supported by IE](https://caniuse.com/#feat=template-literals).
3. Still requires rollup plugin to remove comments.
4. Deviates from glslify format.

**Comment by Mugen87** [comment]
@arodic Sorry for this question, I'm not sure this was already discussed. But why not using [three.module.js](https://github.com/mrdoob/three.js/blob/dev/build/three.module.js) like in this example?

https://jsfiddle.net/f2Lommf5/10743/

**Comment by Itee** [comment]
@arodic Did you take look at [three-full](https://github.com/Itee/three-full) ? All shaders are put in the same folder and are es6 modules. In that way you should be able to import them in the way you want. Waiting that [#9562](https://github.com/mrdoob/three.js/issues/9562) be close.
